File: api/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/service/gsak/SakerService.java
package no.nav.sbl.dialogarena.modiabrukerdialog.api.service.gsak;

import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.gsak.Sak;
import no.nav.sbl.dialogarena.modiabrukerdialog.api.exceptions.JournalforingFeilet;

import java.util.ArrayList;
import java.util.List;

public interface SakerService {
    class Resultat {
        public List<Sak> saker;
        public List<String> feiledeSystemer;

        public Resultat() {
            this(new ArrayList<>(), new ArrayList<>());
        }

        public Resultat(List<Sak> saker, List<String> feiledeSystemer) {
            this.saker = saker;
            this.feiledeSystemer = feiledeSystemer;
        }
    }

    Resultat hentSaker(String fnr);

    @Deprecated
    List<Sak> hentSammensatteSaker(String fnr);

    @Deprecated
    List<Sak> hentPensjonSaker(String fnr);

    void knyttBehandlingskjedeTilSak(String fnr, String behandlingskjede, Sak sak, String enhet) throws JournalforingFeilet;
}

File: api/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/service/gsak/SakerService.kt
package no.nav.sbl.dialogarena.modiabrukerdialog.api.service.gsak

import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.gsak.Sak
import no.nav.sbl.dialogarena.modiabrukerdialog.api.exceptions.JournalforingFeilet


interface SakerService {
    class Resultat(val saker: ArrayList<Sak> = ArrayList(), val feiledeSystemer: ArrayList<String?> = ArrayList())

    fun hentSaker(fnr: String): Resultat

    @Deprecated("")
    fun hentSammensatteSaker(fnr: String): List<Sak>

    @Deprecated("")
    fun hentPensjonSaker(fnr: String): List<Sak>

    @Throws(JournalforingFeilet::class)
    fun knyttBehandlingskjedeTilSak(fnr: String?, behandlingskjede: String?, sak: Sak, enhet: String?)
}
